subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformExecutionBuildOperationIntegrationTest.groovy

```diff
@@ -176,6 +205,371 @@ class ArtifactTransformExecutionBuildOperationIntegrationTest extends AbstractIn
+                }
+            }
+        """
+
+        when:
+        withBuildCache().fails(":consumer:resolve")
+        then:
+        def projectExecution = findTransformExecution { it.artifactName == 'producer.jar' }
+        projectExecution.failure == "java.lang.RuntimeException: BOOM"
+        with(projectExecution.result) {
+            skipMessage == null
+            originExecutionTime == null
+            originBuildInvocationId == null
+            executionReasons == ['No history is available.']
+            cachingDisabledReasonMessage == null
+            cachingDisabledReasonCategory == null
+        }
+        with(snapshotInputsOperation(projectExecution).result) {
+            hash != null
+            classLoaderHash != null
+            implementationClassName == 'MakeGreen'
+            inputValueHashes.keySet() ==~ ['inputArtifactPath', 'inputPropertiesHash']
+            outputPropertyNames == ['outputDirectory', 'resultsFile']
+            inputFileProperties.keySet() ==~ ['inputArtifact', 'inputArtifactDependencies']
+            with(inputFileProperties.inputArtifact) {
+                hash != null
+                attributes == ['DIRECTORY_SENSITIVITY_DEFAULT', 'FINGERPRINTING_STRATEGY_RELATIVE_PATH', 'LINE_ENDING_SENSITIVITY_NORMALIZE_LINE_ENDINGS']
+                roots
+            }
+        }
+        actionExecution(projectExecution) != null
+    }
+
+    def "classpath notation #classpathNotation transforms can be captured"() {
+        settingsFile << """
+            include 'producer', 'consumer'
+        """
+
+        setupBuildWithColorTransform()
+        buildFile << """
+            abstract class MakeGreen implements TransformAction<TransformParameters.None> {
+                @InputArtifact
+                abstract Provider<FileSystemLocation> getInputArtifact()
+
+                void transform(TransformOutputs outputs) {
+                    def input = inputArtifact.get().asFile
+                    println "processing [\${input.name}]"
+                    assert input.file
+                    def output = outputs.file(input.name + ".green")
+                    output.text = input.name + ".green"
+                }
+            }
+        """
+
+        // We need to add a Kotlin build file here so we have access to `gradleKotlinDsl()`
+        file("consumer/build.gradle.kts") << """
+            val color = Attribute.of("color", String::class.java)
+            dependencies {
+                implementation($classpathNotation)
+
+                artifactTypes {
+                    create("jar") {
+                        attributes.attribute(color, "blue")
+                    }
+                }
+            }
+        """
+
+        when:
+        withBuildCache().run ":consumer:resolve"
+        then:
+        def executionIdentifications = buildOperations.progress(IdentifyTransformExecutionProgressDetails).details
+
+        executionIdentifications.each { with(it) {
```

Comment:
🎉  It's a cool pattern!

I wonder if IDE would reformat it into two lines, if you just do a reformat of the whole file. If yes, I wonder how could we avoid that. I would love all sources files on gradle/gradle to be fix points of formatting in IDE

Reply:
The GE build has a `verifyAll` method that encapsulates this even a little bit nicer. Maybe we should do something similar on our side.